processing/src/test/java/io/druid/segment/incremental/OnheapIncrementalIndexBenchmark.java

```diff
@@ -202,12 +208,15 @@ protected AddToFactsResult addToFacts(
 
 
         // Last ditch sanity checks
-        if (numEntries.get() >= maxRowCount && getFacts().getPriorIndex(key) == IncrementalIndexRow.EMPTY_ROW_INDEX) {
-          throw new IndexSizeExceededException("Maximum number of rows reached");
+        if (numEntries.get() >= maxRowCount
+            || sizeInBytes.get() >= maxBytesInMemory
```

Comment:
Looks that it should be 

```
if ((numEntries.get() >= maxRowCount || sizeInBytes.get() >= maxBytesInMemory)
  && getFacts().getPriorIndex(key) == IncrementalIndexRow.EMPTY_ROW_INDEX) {
```

Reply:
Ah yeah, this is a good point. In general `x || y && z` stuff is hard to understand without parentheses.

Reply:
Not only the line break, but also the parentheses are missing.